### PR TITLE
[conda doctor] Don't display `.pyc` and `.pyo` files in "missing files" health check

### PR DESCRIPTION
**Pull request by ForgottenProgramme**
### Description
resolves https://github.com/conda/conda/issues/13370

This PR adds a check to avoid adding any `.pyc` files to the output of the "missing files" health check. 
Missing `.pyc` files can be safely ignored, and by reporting these missing files `conda doctor` unnecessarily alarms the user. 

### To Do

- [x] Add an extra check for missing files ending with `.pyc` extension. Don't add these files to the `packages_with_missing_files` dictionary.
- [x] Test this new behavior.
- [x] Make necessary changes to other existing tests. 
- [x] parameterize the `env_ok` fixture

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

File: conda/plugins/subcommands/doctor/health_checks.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ def find_packages_with_missing_files(prefix: str | Path) -> dict[str, list[str]]
     for file in (prefix / "conda-meta").glob("*.json"):
         for file_name in json.loads(file.read_text()).get("files", []):
             # Add warnings if json file has missing "files"
-            if not (prefix / file_name).exists():
+            if not (prefix / file_name).exists() and not file_name.endswith(".pyc"):
```
Comment (kenodegard) [inline conda/plugins/subcommands/doctor/health_checks.py:66]: Checking if `file_name.endswith(".pyc")` is an in-memory operation (cheap) but checking `(prefix / file_name).exists()` is a disk-based operation (expensive) so we want to do the cheaper operation first so we can benefit from short-circuiting:

```suggestion
            if not file_name.endswith(".pyc") and not (prefix / file_name).exists():
```

**Review by kenodegard — CHANGES_REQUESTED** [review]

File: conda/plugins/subcommands/doctor/health_checks.py (hunk 1 above)
Comment (ForgottenProgramme) [reply conda/plugins/subcommands/doctor/health_checks.py:66]: That makes sense! Thank you.

**Review by kenodegard — CHANGES_REQUESTED** [review]
Need to add a news file, otherwise this is good to go!

File: news/13931-no-add-pyc-doctor [hunk 2]
```diff
@@ -0,0 +1,19 @@
+### Enhancements
+
+* Don't add `.pyc` files to the "missing files" conda doctor health check.
```
Comment (kenodegard) [inline news/13931-no-add-pyc-doctor:3]:
```suggestion
* Skip checking for `.pyc` files in the `conda doctor` "missing files" health check. (#13370 via #13931)
```

**Review by kenodegard — DISMISSED** [review]

**Review by travishathaway — DISMISSED** [review]
🚀

**Comment by jezdez** [comment]
this should look for pyo files as well

**Review by travishathaway — DISMISSED** [review]

**Review by jezdez — CHANGES_REQUESTED** [review]

File: conda/plugins/subcommands/doctor/health_checks.py [hunk 3]
```diff
@@ -56,14 +56,22 @@ def samefile(path1: Path, path2: Path) -> bool:
     return False
 
 
+def excluded_files_check(filename: str) -> bool:
+    excluded_extentions = (".pyc", ".pyo")
+    return filename.endswith(excluded_extentions)
```
Comment (dholth) [inline conda/plugins/subcommands/doctor/health_checks.py:61]:
```suggestion
    excluded_extensions = (".pyc", ".pyo")
    return filename.endswith(excluded_extensions)
```

Comment (ForgottenProgramme) [reply conda/plugins/subcommands/doctor/health_checks.py:61]: ah, spelling mistake! Hihi. 
Thanks, @dholth ! :)

File: news/13931-no-add-pyc-doctor [hunk 4]
```diff
@@ -0,0 +1,18 @@
+### Enhancements
+
+* Skip checking for `.pyc` and `.pyo` files in the `conda doctor` "missing files" health check. (#13370 via #13931)
+### Bug fixes
```
Comment (beeankha) [inline news/13931-no-add-pyc-doctor:4]:
```suggestion
* Skip checking for `.pyc` and `.pyo` files in the `conda doctor` "missing files" health check. (#13370 via #13931)

### Bug fixes
```

File: tests/plugins/subcommands/doctor/test_health_checks.py [hunk 5]
```diff
@@ -29,13 +29,14 @@
     from pytest_mock import MockerFixture
 
 
-@pytest.fixture
-def env_ok(tmp_path: Path) -> Iterable[tuple[Path, str, str, str]]:
+@pytest.fixture(params=[".pyo", ".pyc"])
```
Comment (jezdez) [inline tests/plugins/subcommands/doctor/test_health_checks.py:32]: :chefskiss:

**Review by jezdez — APPROVED** [review]

**Comment by smalltalkman** [comment]
Should the `excluded_files_check` code logic also apply to the `find_altered_packages` function?

**Pull request merged**